time/src/test/java/net/jqwik/api/time/DatesTests.java

```diff
@@ -1,20 +1,342 @@
+
+				assertAllGenerated(dates.generator(1000), random, date -> {
+					assertThat(date.getDayOfMonth()).isGreaterThanOrEqualTo(startDayOfMonth);
+					assertThat(date.getDayOfMonth()).isLessThanOrEqualTo(endDayOfMonth);
+					return true;
+				});
+
+			}
+
+			@Property
+			void dayOfMonthBetweenSame(@ForAll("dayOfMonths") int dayOfMonth, @ForAll Random random) {
+
+				Arbitrary<LocalDate> dates = Dates.dates().dayOfMonthBetween(dayOfMonth, dayOfMonth);
+
+				assertAllGenerated(dates.generator(1000), random, date -> {
+					assertThat(date.getDayOfMonth()).isEqualTo(dayOfMonth);
+					return true;
+				});
+
+			}
+
+			@Provide
+			Arbitrary<Integer> dayOfMonths() {
+				return Arbitraries.integers().between(1, 31);
+			}
+
+		}
+
+		@Group
+		class OnlyDaysOfWeekMethods {
+
+			@Property
+			void onlyDaysOfWeek(@ForAll("onlyDayOfWeeks") DayOfWeek[] dayOfWeeks, @ForAll Random random) {
+
+				Arbitrary<LocalDate> dates = Dates.dates().onlyDaysOfWeek(dayOfWeeks);
+
+				assertAllGenerated(dates.generator(1000), random, date -> {
+					assertThat(date.getDayOfWeek()).isIn(dayOfWeeks);
+					return true;
+				});
+			}
+
+			@Provide
+			Arbitrary<DayOfWeek[]> onlyDayOfWeeks() {
+				return generateDayOfWeeks();
+			}
+
+			Arbitrary<DayOfWeek[]> generateDayOfWeeks() {
+				Arbitrary<DayOfWeek> dayOfWeekArbitrary = Arbitraries
+																  .of(DayOfWeek.MONDAY, DayOfWeek.TUESDAY, DayOfWeek.WEDNESDAY, DayOfWeek.THURSDAY, DayOfWeek.FRIDAY, DayOfWeek.SATURDAY, DayOfWeek.SUNDAY);
+				Arbitrary<Integer> length = Arbitraries.integers().between(1, 7);
+				Arbitrary<List<DayOfWeek>> arbitrary = length.flatMap(depth -> Arbitraries.recursive(
+						() -> dayOfWeekArbitrary.map(v -> new ArrayList<>()),
+						(v) -> addDayOfWeek(v, dayOfWeekArbitrary),
+						depth
+				));
+				return arbitrary.map(v -> v.toArray(new DayOfWeek[]{}));
+			}
+
+			Arbitrary<List<DayOfWeek>> addDayOfWeek(Arbitrary<List<DayOfWeek>> listArbitrary, Arbitrary<DayOfWeek> monthArbitrary) {
+				return Combinators.combine(listArbitrary, monthArbitrary).as(this::addToList);
+			}
+
+			List<DayOfWeek> addToList(List<DayOfWeek> list, DayOfWeek dayOfWeek) {
+				if (!list.contains(dayOfWeek)) {
+					list.add(dayOfWeek);
+				}
+				return list;
+			}
+
+		}
+
+	}
+
+	@Group
+	class Shrinking {
+
+		@Property
+		void defaultShrinking(@ForAll Random random) {
+			DateArbitrary dates = Dates.dates();
+			LocalDate value = shrinkToMinimal(dates, random);
+			assertThat(value).isEqualTo(LocalDate.of(0, Month.JANUARY, 1));
+		}
+
+		@Property
+		void shrinksToSmallestFailingPositiveValue(@ForAll Random random){
+			DateArbitrary dates = Dates.dates();
+			TestingFalsifier<LocalDate> falsifier = date -> date.isBefore(LocalDate.of(2013, Month.MAY, 25));
+			LocalDate value = shrinkToMinimal(dates, random, falsifier);
+			assertThat(value).isEqualTo(LocalDate.of(2013, Month.MAY, 25));
+		}
```

Comment:
How can I fix this?

`Expecting:
     <2014-01-01 (java.time.LocalDate)>
    to be equal to:
     <2013-05-25 (java.time.LocalDate)>
    but was not.`

Or should I accept that it can't shrink perfectly?

Reply:
I'd aim for perfect shrinking here. To see why it won't happen requires detailed analysis. Leave it as is, and create a list with things that don't work as expected yet.

Reply:
List is here: https://github.com/jlink/jqwik/issues/140#issuecomment-743322513